### PR TITLE
stop specifying master branch in optional kubernetes kind presubmits

### PR DESCRIPTION
**Pull request by BenTheElder**
- use kind from artifact
- use decorated job instead of bootstrap
- use skip_branches instead of branches
xref: https://github.com/kubernetes-sigs/kind/issues/2120

### CODE REVIEW

File: config/jobs/kubernetes/sig-testing/conformance-e2e.yaml [hunk 1]
```diff
@@ -70,16 +71,11 @@ presubmits:
         # skip serial tests and run with --ginkgo-parallel
         - name: "PARALLEL"
           value: "true"
-        args:
-        - "--job=$(JOB_NAME)"
-        - "--root=/go/src"
-        - "--repo=k8s.io/kubernetes=$(PULL_REFS)"
-        - "--repo=sigs.k8s.io/kind=master"
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-testing/conformance-e2e.yaml:77]: reported by @jsturtevant https://kubernetes.slack.com/archives/C09QZ4DQB/p1616008878011700

**Comment by jsturtevant** [comment]
lgtm but I am not super familiar with difference between krte and kubekins images.

**Comment by jsturtevant** [comment]
/assign @dims 

this is blocking https://github.com/kubernetes/kubernetes/pull/100337

**Comment by BenTheElder** [comment]
images/krte is used by kind jobs and is a focused subset of kubekins for kind
images/kubekins is used by everything under the sun 🙃
They share a variants.yaml to keep the go version in sync.

**Comment by dims** [comment]
looks like this has all the stuff to merge :)

/approve
/lgtm 

from me as well.

**Pull request merged**